File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/AlmacenService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Almacen;
import es.uvigo.mei.pedidos.entidades.ArticuloAlmacen;
import es.uvigo.mei.pedidos.entidades.Articulo;

public interface AlmacenService {
	public Almacen crear(Almacen almacen);
	public Almacen modificar(Almacen almacen);
	public void eliminar(Almacen almacen);
	public Optional<Almacen> buscarPorId(Long id);
	public List<Almacen> buscarTodos();
	public List<Almacen> buscarPorNombre(String patron);
	public List<Almacen> buscarPorLocalidad(String localidad);
	public List<Almacen> buscarPorArticuloId(Long articuloId);

	public ArticuloAlmacen crearArticuloAlmacen(ArticuloAlmacen articuloAlmacen);
	public ArticuloAlmacen crearArticuloAlmacen(Articulo articulo, Almacen almacen, Integer stock);
	public ArticuloAlmacen modificarArticuloAlmacen(ArticuloAlmacen articuloAlmacen);
	public void eliminarArticuloAlmacen(ArticuloAlmacen articuloAlmacen);
	public void eliminarArticuloAlmacen(Long idArticulo, Long idAlmacen);
	public Optional<ArticuloAlmacen> buscarArticuloAlmacenPorArticuloIdAlmacenId(Long articuloId, Long almacenId);
	public List<ArticuloAlmacen> buscarArticulosAlmacenPorAlmacenId(Long almacenId);
	public List<ArticuloAlmacen> buscarArticulosAlmacenPorArticuloId(Long articuloId);
}

File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/almacenService.js
import api from "./pedidosAPI";

class AlmacenService {
  buscarTodos() {
    return api.get("/almacenes");
  }

  buscarPorId(id) {
    return api.get(`/almacenes/${id}`);
  }

  crear(data) {
    return api.post("/almacenes", data);
  }

  modificar(id, data) {
    return api.put(`/almacenes/${id}`, data);
  }

  eliminar(id) {
    return api.delete(`/almacenes/${id}`);
  }

  buscarPorLocalidad(localidad) {
    return api.get(`/almacenes?localidad=${localidad}`);
  }

  buscarPorArticuloId(idArticulo) {
    return api.get(`/almacenes?articuloId=${idArticulo}`);
  }

  buscarArticulosAlmacen(id) {
    return api.get(`/almacenes/${id}/articulos`);
  }

  anadirArticuloAlmacen(idAlmacen, nuevoArticuloAlmacen) {
    return api.post(`/almacenes/${idAlmacen}/articulos`, nuevoArticuloAlmacen);
  }

  modificarArticuloAlmacen(idAlmacen, idArticulo, nuevoArticuloAlmacen) {
    return api.put(`/almacenes/${idAlmacen}/articulos/${idArticulo}`, nuevoArticuloAlmacen);
  }

  eliminarArticuloAlmacen(idAlmacen, idArticulo) {
    return api.delete(`/almacenes/${idAlmacen}/articulos/${idArticulo}`);
  }
}

const service = new AlmacenService();
export default service;
